extensions/shared/upgrade-nudge/index.jsx

```diff
@@ -0,0 +1,47 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+import { __, sprintf } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+import { Button } from '@wordpress/components';
+import { withSelect } from '@wordpress/data';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+import getSiteFragment from '../get-site-fragment';
+import './store';
+
+import './style.scss';
+
+const UpgradeNudge = ( { feature, plan, planName } ) => (
+	<div className="jetpack-upgrade-nudge">
+		<Gridicon className="jetpack-upgrade-nudge__icon" icon="star" size={ 18 } />
+		<div className="jetpack-upgrade-nudge__info">
+			<span className="jetpack-upgrade-nudge__title">
+				{ sprintf( __( 'Upgrade to %(planName)s', 'jetpack' ), {
+					planName,
+				} ) }
+			</span>
+			<span className="jetpack-upgrade-nudge__message">
+				{ __( 'To make this block visible on your site', 'jetpack' ) }
+			</span>
+		</div>
+		<Button
+			className="jetpack-upgrade-nudge__button"
+			href={ addQueryArgs( `https://wordpress.com/plans/${ getSiteFragment() }`, {
```

Comment:
Tested on .com with Calypso:

I think we need `target="_top"` here because otherwise this page opens in the iframe that loads Gutenberg editor in Calypso. 😀 

<img width="1286" alt="Screenshot 2019-07-02 at 15 46 11" src="https://user-images.githubusercontent.com/87168/60513733-a23c9f00-9ce0-11e9-8056-dd02efc8c12c.png">

Similarly, we will need to pass on the parent URL and not the iframe URL.

Okay to leave as-is now and we'll just get back to this.

Reply:
Looks like the iframe has `calypsoifyGutenberg` object containing some URLs, so we could add the post URL in Calypso in the same object and use it on to checkout flow as a redirect URL.

Reply:
> I think we need `target="_top"` here because otherwise this page opens in the iframe that loads Gutenberg editor in Calypso. 

73810d181